services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/document/EnhancedDocumentTest.java

```diff
@@ -496,4 +496,162 @@ void error_when_usingClassGetPut_for_CollectionValues(){
                                                 () -> EnhancedDocument.builder().build().get("listKey" , List.class))
                                             .withMessage("Values of type List are not supported by this API, please use the getList API instead");
     }
+
+    @Test
+    void toJson_numberFormatting_veryLargeNumbers() {
+        EnhancedDocument doc = EnhancedDocument.builder()
+            .attributeConverterProviders(defaultProvider())
+            .putNumber("longMax", Long.MAX_VALUE)
+            .putNumber("longMin", Long.MIN_VALUE)
+            .putNumber("doubleMax", Double.MAX_VALUE)
+            .putNumber("scientific", 1.23e+100)
+            .putNumber("manyDecimals", 1.123456789012345)
+            .build();
+        
+        String json = doc.toJson();
+        assertThat(json).isEqualTo("{\"longMax\":9223372036854775807,\"longMin\":-9223372036854775808,"
+                                  + "\"doubleMax\":1.7976931348623157E308,\"scientific\":1.23E100,"
+                                  + "\"manyDecimals\":1.123456789012345}");
+    }
+
+    @Test
+    void toJson_numberFormatting_trailingZeros() {
+        EnhancedDocument doc = EnhancedDocument.builder()
+            .attributeConverterProviders(defaultProvider())
+            .putNumber("twoPointZero", 2.0)
+            .putNumber("tenPointZeroZero", 10.00)
+            .build();
+        
+        String json = doc.toJson();
+        assertThat(json).isEqualTo("{\"twoPointZero\":2.0,\"tenPointZeroZero\":10.0}");
+    }
```

Comment:
Comparing JSON like this is brittle; the order of the properties in the output JSON could change for example. Ideally, we should do something like this

```java
JsonNode actual = mapper.readTree(doc.toJson());
JsonNode expected = mapper.readTree("{\"twoPointZero\":2.0...);

assertThat(actual).isEqualTo(expected);
```

Reply:
done